moveit_core/robot_state/include/moveit/robot_state/robot_state.h

```diff
@@ -1495,6 +1495,15 @@ as the new values that correspond to the group */
    */
   bool isValidVelocityMove(const RobotState& other, const JointModelGroup* group, double dt) const;
 
+  /**
+   * \brief Check that the time to move between two waypoints is sufficient given velocity limits
+   */
+  bool isValidVelocityMove(const JointModelGroup* group, const std::vector<double>& from_joint_pose,
+                           const std::vector<double>& to_joint_pose, double dt) const;
+
+  bool isValidVelocityMove(const JointModelGroup* group, const double* from_joint_pose,
+                           const double* to_joint_pose, std::size_t array_size, double dt) const;
+
```

Comment:
why the overload with raw pointers? Convenience?

Reply:
speed! this is used in a more MoveIt!-integrated version of Descartes, where it is called hundreds of thousands of times in an iterative IK brute force-type approach. It uses Boost-graph to save each raw array of joints

Reply:
Passing a reference is slower than passing a pointer? Afaik they are basically the same, and the difference is purely syntactical.

Reply:
Oh I thought you were asking why these two functions instead of the RobotState version. Its overloaded just as a convenience I suppose, I don't recall why I have both, I'm just trying to get my research code to work with mainstream moveit again and not my fork. I think this style is not unfounded within RobotState however - e.g. [setJointPosition()](https://github.com/ros-planning/moveit/blob/kinetic-devel/moveit_core/robot_state/include/moveit/robot_state/robot_state.h#L483) has it overloaded.

Reply:
robot state uses both vectors and raw pointers for lots of its functions, so i think this should remain.